### PR TITLE
Improve e2e error reporting

### PR DESCRIPTION
**Pull request by serathius**
cc @fuweid @jmhbnz @MadhavJivrajani @siyuanfoundation

### CODE REVIEW

File: pkg/expect/expect.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func (ep *ExpectProcess) ExpectFunc(ctx context.Context, f func(string) bool) (s
 
 		select {
 		case <-ctx.Done():
-			return "", fmt.Errorf("failed to find match string: %w", ctx.Err())
+			return "", fmt.Errorf("context done before to found matching log")
```
Comment (jmhbnz) [inline pkg/expect/expect.go:196]: Sentence is a bit hard to understand, perhaps:
```suggestion
			return "", fmt.Errorf("context done before matching log found")
```

**Comment by serathius** [comment]
Fixed the comments

**Pull request merged**